core/mo-services-impl/nmf-software-management-impl/src/main/java/esa/mo/sm/impl/provider/AppsLauncherManager.java

```diff
@@ -396,6 +407,9 @@ protected boolean isAppRunning(final Long appId)
       ret.add(str.toString());
     } else {
       if (runAs != null) {
+        if(sudoAvailable){
```

Comment:
If sudo is available, `sudo su` is an antipattern as it escapes the sudoers permission lists - i.e. using `sudo` + sudoers configuration is the proper way to ensure that supervisor is not running as a privileged user, and thus commands like
`su - user123 -c <command>` should be replaced with `sudo -u user123 <command>`. This along with properly tailored system config (i.e. ensuring that `user123` belongs to a group which is sudo'able by the supervisor user) will properly utilise linux permissions system.

Reply:
This is the thorny one... while I agree with you, there was some problem which prevented me from doing what you are suggesting, can't quite recall it however

Reply:
Please give it a go. Message me if it gives you issues. Perhaps adding `-i` switch to sudo is enough in case of problems.